### PR TITLE
Branch anuj

### PR DESCRIPTION
**Pull request by AnujUTS**

### CODE REVIEW

File: backend/src/main/java/com/example/subjecthub/controller/FacultyController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @RequestMapping(value = "", method = RequestMethod.GET)
+    public List<Faculty> getFaculties(
+        @PathVariable Long universityId,
+        @RequestParam(required = false) String name,
+        @RequestParam(required = false) String code
+    ) {
+        return facultyRepository.findByUniversityId(universityId).stream()
+            .filter(s -> (name == null||s.getName().equalsIgnoreCase(name)))
+            .filter(s -> (code == null||s.getCode().equalsIgnoreCase(code)))
```
Comment (yaseen95) [inline backend/src/main/java/com/example/subjecthub/controller/FacultyController.java]: Not sure if you recall. But s.getCode() can return null, so calling .equalsIgnoreCase might throw an NPE. E.g. api/universities/university/2/faculties?code=aklsdfjadsklf throws an error because UOW's faculties do not have codes.

To avoid this, switching the order might help, i.e.
```java
code.equalsIgnoreCase(s.getCode())
```

I'd like you to add a test where you fetch a faculty that has a code that can be null to ensure that this doesn't throw an exception.
